Clarify fully specified vs generic

poeschlr opened the pull request:
* Clarified KiCad footprint assignment work-flows as generic vs fully specified symbols (G2.1)
  * Avoid using the word atomic as it does not really specify what is really used.
  * Added a small definition where generic symbols are to be found

This is my second attempt at clarifying this. this time i made a concussion effort not to change anything of substance but manly clarify what we mean with fully specified and most importantly avoided the word atomic.

poeschlr commented on content/klc/G2.1.adoc:
```diff
@@ -1,21 +1,29 @@
 +++
 
-In general terms, the _symbols_ in the KiCad library can be categorized into two groups:
+KiCad generally has two work-flows with regards to assign a footprint to a symbol. It can be done in the library or using the "assign footprints tool" at the end of the design process.
```
The symbol does not need to be created by the user. It can be from the official lib for example.

poeschlr replied:
Follow the community for a while and you will notice there are two camps. These two do not go along well and this is the reason for describing both workflows without implying one to be superior.

evanshultz commented:
@poeschlr 
Could this be expanded to point out that only a few limited libraries are allowed to have no default footprint ("Generic symbols" in this context)?

poeschlr commented:
Is already in there:
> **Where to use which type**
>
>The official library generally uses fully specified symbols where feasible. Generic symbols are limited to a small number of libraries that contain only generic symbols. These generic libraries are:
>
>* `Device`
>* All `Connector` libraries
>* The logic family libraries like `74xx`, `4xxx`, ...

evanshultz commented:
Thank you. It is the use of "generally" and "feasible" that make it read loosely, to me.

May I suggest this?
```
The official library requires fully specified symbols except for in a small number of libraries that contain only generic symbols. These generic libraries are:
```

poeschlr commented:
Upadated wording and fixed missconception about "workflow"

I also discovered that S2.2 repeated and partly contraticted what i added so i limited the skope of that rule to only talk about wildcarding non functional part number options.

I clarified the use of clear text package suffixes after the common base part number to be limited to parts build by multiple manufacturers.

evanshultz commented on content/klc/S2.1.adoc:
```diff
@@ -4,11 +4,12 @@ brief = "General symbol naming guidelines"
-. Specific manufacturer name (for fully specified symbols)
-. Type of symbol (for generic symbols)
+. Fully specified symbols are named based on the manufacturer part number
+.. If multiple manufacturers build compatible devices then the common base part number is used with clear name suffix for the footprint (e.g. `L78L05_TO92`)
```
Perhaps change "compatible" to "interchangeable" so show an exact equivalence?

Perhaps change "used with clear name suffix for the footprint" with "suffixed with the footprint type"?

poeschlr replied:
No if one manufacturer builds a device then we use the full part number. The suffix is a workaround for manufacturers using different part numbers for the same exact device.

matthijskooijman replied:
Ah, I see. Perhaps make that explicit? e.g. `.. Since the full MPN already implies the footprint of the part, there is no need to add it to the name. But if multiple manufacturers build compatible devices (but with different designators for the footprint) then the common base part number is used with clear name suffix for the footprint (e.g. `L78L05_TO92`)`

poeschlr replied:
That is way too long for this set of rules. Really a bit of common sense is to be expected by users. i will simply replace the if with if and only if (short iff)

evanshultz commented on content/klc/S2.1.adoc:
```diff
@@ -4,11 +4,12 @@ brief = "General symbol naming guidelines"
-. Type of symbol (for generic symbols)
+. Fully specified symbols are named based on the manufacturer part number
+.. If multiple manufacturers build compatible devices then the common base part number is used with clear name suffix for the footprint (e.g. `L78L05_TO92`)
+.. Non functional parts of the MPN are replace with a wildcard. (Refer to   link:/libraries/klc/S2.2[requirements for the use of wildcards in MPN])
```
Are the multiple spaces after "Refer to" intentional?

evanshultz commented:
This is looking good to me. Thanks! I made a couple more comments to consider.

poeschlr commented on content/klc/G2.1.adoc:
```diff
@@ -1,21 +1,29 @@
 
 {{< klcimg src="G2.1c" title="Atomic symbol" >}}
 
+Some users refer to fully specified symbols as atomic parts or a library of such components as atomic library. Atomic implies a one to one inseparable connection between assets. The closest that KiCad can get to is a fully specified symbol that points to a footprint specialized to the same component.
```
Is in the KLC history explanation. This is where history belongs.

matthijskooijman replied:
Ok

poeschlr commented on content/klc/S2.1.adoc:
```diff
@@ -4,11 +4,12 @@ brief = "General symbol naming guidelines"
 .. Reference designator may be substituted for common components (e.g. `D`, `C`, `LED`)
-. Part name should include extension for specific footprint if required (e.g. `SOIC`)
 . Any modification of the original symbol, indicated by appending the reason (e.g. different pin ordering - `Q_NPN_CBE`, `Q_NPN_BCE`)
 . Indicate quantity of elements for symbol arrays (e.g. resistor array with 8 elements - `Resistor_x8`)
```
Not part of this pull request, to be ignored here

poeschlr replied:
Oh i misread. The modification and quantity will most likely be used in a generic symbol but i would for now keep it as top level. (I do not want to restrict us too much)

matthijskooijman replied:
Ok

evanshultz replied:
I don't think so since these are all symbol name options which can be all mutually-exclusive.

poeschlr commented on content/klc/S2.2.adoc:
```diff
@@ -13,7 +13,3 @@ Examples:
 To capture every possible combination of part variation requires a large number of symbol aliases - this is to be avoided.
```
I do not want to go into too much detail with the KLC as it is already too long. the fact that different footprints require different symbols is made abandonedly clear in other rules and does not need to be repeated here. The rest will highly depend on device type and is something that will mostly be dealt with in the pull requests. (I could however mention that temperature ranges, standard compliance and packaging variants should never be included.)

matthijskooijman replied:
I wouldn't consider the KLC too long (it's a long read, but when you have a specific question about drawing a part, details and examples are always helpful). In this case, there is a distinction made and having examples of both side of the distinction could help to make the distinction more clear (as well as provide more clear rules for any parts to which the examples apply directly). But I won't insist more than this :-)

evanshultz replied:
When a single manufacturer makes a part in one or more packages, the MPN is used. No generic suffix of the package type is required because there will be no conflicts.

Just above this line you selected in S2.2 are examples of things to wildcard. Footprints are naturally captured by the symbol and that's already covered. Voltage ratings are usually split out into ALIASes these days instead of wildcarding as per Rene's preference and to make part selection easier as those are functional differences. I don't know about accuracy but I would be inclined to wildcard it since that only limits the range at possible values and is usually a binned/tested characteristic, and not a design/functional difference.

poeschlr commented on content/klc/G2.1.adoc:
```diff
@@ -1,21 +1,29 @@
+**Fully specified symbols**
 
-Atomic parts fully define a component, specifying a matching footprint, and are named based on the `MPN` (manufacturer part number). Atomic parts are ready to be placed onto the PCB as they are already associated with a footprint.
+Fully specified symbols define a matching footprint, and are named based on the `MPN` (manufacturer part number). Such symbols do not depend on the user to select the correct footprint later but can have alternative footprints defined using footprint filters such that the user can define it later.
```
Is specified via the rules for footprint filters (Does not need to be repeated here)

matthijskooijman replied:
Ah, indeed.

evanshultz replied:
@poeschlr 
How about changing the last words of "can define it later" to "can select an alternate footprint"? A footprint is already defined and the footprint filters only allow alternate footprint selection.

poeschlr commented on content/klc/S2.1.adoc:
```diff
@@ -4,11 +4,12 @@ brief = "General symbol naming guidelines"
-. Part name should include extension for specific footprint if required (e.g. `SOIC`)
 . Any modification of the original symbol, indicated by appending the reason (e.g. different pin ordering - `Q_NPN_CBE`, `Q_NPN_BCE`)
 . Indicate quantity of elements for symbol arrays (e.g. resistor array with 8 elements - `Resistor_x8`)
 . Add suffixes for symbol variations
```
Yes. See next line in the file.

matthijskooijman replied:
Ok

evanshultz replied:
You've highlighted the line "Part name should include..." and that applies for parts which have different suffixes for the same package but are physically identical. This let's us avoid having multiple identical symbols for interchangeable parts from different manufacturers (search for symbols which include `_SOIC` for examples). I've hopefully clarified this at https://github.com/KiCad/kicad-website/pull/417.

matthijskooijman replied:
My original comment referred to the line directly above it, i.e. "Add suffixes for symbol variations".

poeschlr replied:
There was a duplication in the ruleset. Fixed

evanshultz commented on content/klc/S2.1.adoc:
```diff
@@ -4,11 +4,12 @@ brief = "General symbol naming guidelines"
 
 . Library naming should not be duplicated in footprint name
```
Good catch. PR submitted at https://github.com/KiCad/kicad-website/pull/417.

poeschlr commented on content/klc/G2.1.adoc:
```diff
@@ -1,21 +1,29 @@
 **Generic symbols**
 
-Generic symbols can be used with multiple footprints, and _do not_ have a default footprint assigned. Generic symbols allow flexibility in the design workflow. Symbols are first added to the schematic, and the footprint is selected before PCB layout. Using generic symbols allows a small number of library symbol elements to represent a _very large number_ of possible component combinations.
+Generic symbols can be used with multiple footprints, and _do not_ have a default footprint assigned. Generic symbols allow flexibility in the design work-flow. Symbols are first added to the schematic, and the footprint is selected before PCB layout. Using generic symbols allows a small number of library symbol elements to represent a _very large number_ of possible component combinations.
```
I rewrote this section to make more sense.

poeschlr commented:
I think i have addressed everything. @evanshultz could you take a look?

matthijskooijman commented on content/klc/S2.3.adoc:
```diff
@@ -1,21 +1,18 @@
-Many electronic components are provided in multiple footprint (package) options. These _may or may not_ be pin compatible.
+Many electronic components are provided in multiple packaging variants. These _may or may not_ be pin compatible.
+
+KiCad is not able to assign different footprints for aliases. Fully specified symbols therefore require a separate symbol for every package variant.
```
Great addition!

matthijskooijman commented on content/klc/S2.2.adoc:
```diff
@@ -4,12 +4,12 @@ brief = "Non-functional variations in part number should be replaced with wildca
+To capture every possible combination of part variation would require a large number of symbol aliases - this is to be avoided.
 
-Where an MPN has options for such non-functional variations, these portions of the MPN should be replaced with a wildcard character (`x`).
+Where an MPN has options for such non-functional variations, these portions of the MPN should be replaced with a wildcard character (`x`). If these non-functionl parts of the MPN are at the ond of it then these have to be removed instead.
```
s/ond/end/

Also, maybe "omitted" is better than "removed"?

evanshultz commented:
Nick is right. I'm re-reading now.

poeschlr commented:
I did increase the klc version. See https://github.com/KiCad/kicad-website/pull/407/files#diff-88635b43085423a1810bd36ac8b0bc17R50

Or do i misunderstand your request?

evanshultz commented:
The NC pins PR, which looks arguably closer to merging, is 3.0.21. So if this goes in next it will need to be 3.0.22. Or maybe I misunderstood Nick as well.

evanshultz commented on content/klc/G2.1.adoc:
```diff
@@ -1,21 +1,29 @@
 +++
 
-In general terms, the _symbols_ in the KiCad library can be categorized into two groups:
+KiCad generally has two workflows with regards to assign a footprint to a symbol. It can be done in the library or using the "assign footprints tool" at the end of the design process.
```
Remove 'generally'. The number of workflows, at least those defined below, are a known integer number. Change 'with regards to assign' to 'for assigning'. The name of the formerly CvPcb window is just "Assign Footprints" so the word 'tool' can move outside the double quotes. I would add 'schematic' before 'design process'.

poeschlr replied:
completely reworded.

evanshultz replied:
OK. I only ask to change `"assign footprints tool"` to `"Assign Footprints" tool` for reasons mentioned above.

poeschlr replied:
done

evanshultz commented on content/klc/G2.1.adoc:
```diff
@@ -1,21 +1,29 @@
 **Generic symbols**
 
-Generic symbols can be used with multiple footprints, and _do not_ have a default footprint assigned. Generic symbols allow flexibility in the design workflow. Symbols are first added to the schematic, and the footprint is selected before PCB layout. Using generic symbols allows a small number of library symbol elements to represent a _very large number_ of possible component combinations.
+Generic symbols can be used with multiple footprints, and _do not_ have a default footprint assigned. Generic symbols allow flexibility in the design workflow. For this workflow the symbol placed into the schematic is typically selected to fit the intended function. The footprint assignment happens later when the exact part number is selected. (At the latest when starting with the layout) Using g ...
```
Missing a period for the sentence in quotes. And maybe that sentence isn't even needed?

matthijskooijman replied:
Do you mean the sentence in parenthesis?

> (At the latest when starting with the layout)?

I think that it would be proper for the period before to be moved after it, so:

> The footprint assignment happens later when the exact part number is selected (At the latest when starting with the layout).

But I also wonder if the part in parenthesis is really needed, leaving it out might be better?

poeschlr replied:
removed that part and reworded section

evanshultz commented on content/klc/G2.1.adoc:
```diff
@@ -1,21 +1,29 @@
 
 {{< klcimg src="G2.1c" title="Atomic symbol" >}}
 
+Some users refer to fully specified symbols as atomic parts or a library of such components as atomic library. Atomic implies a one to one inseparable connection between assets. The closest that KiCad can get to is a fully specified symbol that points to a footprint specialized to the same component.
```
This implies to me there is a limitation of KiCad, when it's really the system defined by KLC. A fully atomic library would be possible in KiCad. Can this be re-worded to solely focus on the library resulting from the rules defined in KLC?

poeschlr replied:
The inseparatable part is a limitation of kicad as the symbol and footprint libs are indeed separate. (Even if a symbol points to a footprint there is nothing that ensures that the same footprint is not used by another symbol. A true atomic entity would ensure this. At least if you go by the strictest definition of the word as used for example when talking about atomic operations of a CPU.)

evanshultz replied:
Umm...

Anyway, maybe I should say I like if the note about KiCad is removed and instead the doc just focuses on the library system defined by KLC. You are contrasting the perception of the word 'atomic' to the terms and library given by following KLC. I guess it seems subjective and interpretive to me instead of just getting down to business and stating what KLC is intended to accomplish in this section.

Please add 'an' before 'atomic library' in the first sentence.

poeschlr replied:
The reason why this section even exists is because the terms used are sadly not standardized. Every user has a slightly different interpretation of them depending on which field or software they come from. So i really need to define the terms and how exactly they are used in the context of the KLC as it will otherwise lead to misinterpretation of the KLC.
As far as i understand it altium for example has true atomic libs which is why i also highlight why it is not possible with kicads data structure. I can remove this one part (the stuff about why it is not possible to make true atomic libs) but i am not prepared to remove the definition of terms.

poeschlr replied:
I massively reworded this section and made it cursive to make it clear that this is a sidenote (that mainly deals with the confusing use of atomic symbols and atomic parts to refer to very different things.)

evanshultz replied:
Alright. Please add a comma after `Confusingly` and we'll call it good.

poeschlr replied:
done

poeschlr commented:
Ah i think i understand now. I updated the main version string to the wrong number. Fixed now.

Should the other pull request indeed be merged first then there will be a merge commit that will then handle updating of the version number one more time.

evanshultz commented on content/klc/S2.1.adoc:
```diff
@@ -2,15 +2,15 @@
-. Specific manufacturer name (for atomic parts)
-. Type of symbol (for generic parts)
+. Fully specified symbols are named based on the manufacturer part number
+.. If and only if multiple manufacturers build compatible devices then the common base part number is used with clear name suffix for the footprint (e.g. `L78L05_TO92`)
```
If the MPNs are identical then there's no problem. This only applies if the MPN suffix for a compatible part with the same MPN and package have conflicts, like 'D' and 'M' suffixes both being for compatible SOIC packages by some IC vendors. So this can be clarified that only when the MPN, including the suffix to define the package, is in conflict does this system need to be employed.

poeschlr replied:
Do you really want to be responsible to check every possible manufacturer? With the rule as stated we can simply say "this is produced by many manufacturers who in theory could use different numbering schemes so please use cleartext"

evanshultz replied:
What I am trying to say is that if the MPN is identical between vendors then this suffix of package isn't needed. We could go with `L78L05Z` and not `L78L05_TO92`, for example, if all vendors use the suffix 'Z' for the TO92 package. But if the functionally- and package-equivalent part is sold as `L78L05A` and `L78L05B` then we use this scheme.

Only if we are alerted to this issue then we should go with this scheme but it doesn't need to be expected. I don't intend to intimate that librarians must scour the internet for every possible PR. We have seen this issue come up just in a few places like 'classic' regulators and opamps/comparators, so we probably know when it's a possibility. I only mean to try and limit the use of this rule as much as possible.

chschlue replied:
Generally, I'm with @poeschlr on this, compatible parts that even share package suffixes among manufactures are probably a rare find so clear text is the safe option.

But is an 78xx really the best example here? It is probably one of the most well known parts in the world but has five dozen manufacturers with ten dozen different part numbers each (different prefixes, different infixes vor low dropout variants, different suffixes for output tolerance, different packages and whatnot)

I can't think of a good one right know, but perhaps at least another (counter-) example should be added.

poeschlr replied:
Here we have i think a different philosophy. For now i will go with your wording as i guess this is how it is done right now (You are after all more active on the symbol side). But i will for sure open an issue about it for discussing if this is the best approach. (Listing pros and cons of both approaches such that we can then select which one to use.)

evanshultz replied:
@chschlue
https://github.com/KiCad/kicad-symbols/pull/1081 is one example. `DG419` is another where ADI uses the suffix `N` and Maxim uses `J` (our library uses a `_Maxim` suffix which isn't described here at all so it may be deprecated). Or `TL072` where ST uses `D`, TI uses `DR`, and Diodes Inc uses `SG`. It's really those common ICs where I've seen it. (Note that when scanning through the library I think this might be mis-used and we don't really need as many examples, or what is there is to maintain compatibility with manufacturers that don't actively make the parts anymore, like `MC1408`.)

@poeschlr 
I would just like to replace `build compatible devices` with `build compatible devices with differing suffixes` to reduce this rule's use as much as possible.

poeschlr replied:
Is already written this way at the point of this answer.

evanshultz replied:
I feel so annoying about continuing with this topic, but I still have a couple of comments.
1. The word `compatible` isn't quite right to me since it gives the impression of "similar, but not necessarily identical". We want to invoke this rule only when the "base" MPN is identical, right?
1. Also, the phrase about describing the suffix seems overwrought to me.

May I suggest `If multiple manufacturers produce equivalent parts where the part number is the same except for the suffix that specifies the package, then the common base part number is kept and suffixed with the footprint name (e.g. `L78L05_TO92`)`?

https://github.com/KiCad/kicad-symbols/pull/2076 is a recent example using parts from datasheets at http://www.ti.com/lit/ds/symlink/sn74lvc1g66.pdf and https://assets.nexperia.com/documents/data-sheet/74LVC1G66.pdf.

matthijskooijman replied:
That proposal looks better to me, too.

chschlue replied:
@evanshultz 
I objected to the `L78L05_TO92` example before but your suggested wording transforms it from questionable to flat out wrong.

evanshultz replied:
I think there are better examples too, but I honestly didn't look into the section into the parenthesis very much. I was trying to make the sentence read better and be specific.

Could you please enlighten us about my wrongness? Maybe with a suggestion of your own? Perhaps you object to the use of `footprint name` before the open paren and would prefer `simplified footprint name`? Re-reading it that sounds better to me.

chschlue replied:
I'm not opposed to the suggested sentence. The problem is: The example contradicts it because equivalent parts all differ at least in prefix, too. Correct me if I'm wrong, but AFAIK an L78* is invariably manufactured by ST. Equivalent parts use MC78*, µA78*, ...
You gave better examples yourself, TL072, DG419, ...

I get that the "base" MPN of the example is 78[Vout], but it's at least misleading when the rule explicitly says package _suffix_ only.

poeschlr replied:
Meaning my original wording fits the current lib better as the `L78L05_TO92` is in our lib.

chschlue replied:
Chicken and egg... all the other examples are already in the lib, sporting wrong names.

evanshultz replied:
Hmm... That's not how I interpret the conversation, Rene.

I've suggested different wording. Matthis concurred. Christian is OK with the wording too, but thinks the example given is a poor one. I agree, but from a different perspective.

Would changing the sentence as shown above and _also_ changing the example above satisfy everyone? What objection is there to that idea? I originally noted this paradigm at the link below for `MC33078`. `DG419` and `TL072` are also noted above.

chschlue replied:
LF398 would work in it's current state. Caveat: TI seems to be only manufacturer left.

chschlue replied:
Oh, I wasn't aware this might be unclear. Short version:
Yes, I like Evan's wording better.
Yes, I think the current example is bad.
No, I don't have an optimal example that doesn't conflict with the current lib.

poeschlr replied:
The problem is that the change in KLC was intended to capture the current state of the lib. Changing the example would not change the fact that this symbol is in the library. If it contradicts the rule as an example then it also contradicts it just by existing in the lib. So we have two options. Word the rule such that we have the wiggle room such that this symbol is correct or strengthen the wording as suggested and make this symbol non compliant.

chschlue replied:
You are right in that this is definitely a color-of-the-bikeshed discussion. I certainly don't mean to block the fully specified vs. generic stuff over a `fucking example`.

But a more general question: Is KLC really supposed to capture the current state of the library, which is a little rough around the edges to say the least?
Shouldn't the library convention be the set point rather than the process value, if you'll excuse the control loop analogy?

poeschlr replied:
You are right the KLC should point to a future goal.

Here it is however meant to clarify an already existing rule. That was in part created exactly for covering the part listed in the example.

The pull request that added this type of suffix to it was one of the first that i reviewed after evan and jan made this suggestion in the first place. So to me the fact that it now violates the rule shows that the current wording does not cover the original intent and would therefore require more than a change in the minor release number.

Also with the stronger wording is there even a significant number of symbols left where this would be allowed? Because maybe a better future goal (for v6) would be to get rid of this exception and require fully specified with aliases for differing manufacturer conventions. (This would be even easier for us than my wording with wiggle room as we then don't even need to check if multiple manufacturers make this part.)

chschlue replied:
Now I've got it. Probably. Hopefully.
Keep the wording and the fucking example. Nothing to see here. Please disperse.

Is there a clear cut compilation of what is set in stone / planned / wishful thinking for v6 anywhere?
I wasn't able to find much, but maybe I'm just too dumb for Launchpad.

poeschlr replied:
In this case it has nothing to do with what the devs do but it is simply a matter of fact that massive renaming of symbols (which the changed wording seems to require) can simply not happen during a minor release. So it would really need to wait till v6.

---

Really i don't get why this discussion is so heated here. You guys wanted me to make the wording stronger, then you discovered that the new wording (suggested by you) contradicts the example that has always been in this rule and now you are mad at me for pointing this out.

evanshultz replied:
@chschlue
I may be too dumb for Launchpad as well. I also often can't find what I want using the website to search the dev mailing list archive.

evanshultz commented on content/klc/S2.1.adoc:
```diff
@@ -2,15 +2,15 @@
-. Type of symbol (for generic parts)
+. Fully specified symbols are named based on the manufacturer part number
+.. If and only if multiple manufacturers build compatible devices then the common base part number is used with clear name suffix for the footprint (e.g. `L78L05_TO92`)
+.. Non functional parts of the MPN are replace with a wildcard. (Refer to link:/libraries/klc/S2.2[requirements for the use of wildcards in MPN])
```
'Non-functional' is missing the hyphen. 'replace' should be past tense or the entire sentence re-worked to change tense.

poeschlr replied:
done

evanshultz commented on content/klc/S2.2.adoc:
```diff
@@ -4,16 +4,12 @@ brief = "Non-functional variations in part number should be replaced with wildca
 * RoHS / PbFree information
 
-To capture every possible combination of part variation requires a large number of symbol aliases - this is to be avoided.
+To capture every possible combination of part variation would require a large number of symbol aliases - this is to be avoided.
```
It would be more inline with the rest of KLC's wording to avoid a hyphen here and change '- this' to 'and'.

poeschlr replied:
done

evanshultz commented on content/klc/S2.2.adoc:
```diff
@@ -4,16 +4,12 @@ brief = "Non-functional variations in part number should be replaced with wildca
-**Rule of thumb**
-
-A single symbol should be drawn for each footprint variation of an atomic part. Other variations (as listed above) do not require their own symbol alias.
+Where an MPN has options for such non-functional variations, these portions of the MPN should be replaced with a wildcard character (`x`). If these non-functionl parts of the MPN are at the end of it then these have to be omitted.
```
'at the end of it' isn't clear; what is being referred to because it's not the MPN which gets a wildcard. Change 'it then these have to' to 'the symbol name then the wildcard should'? Also there is a missing 'a' in the second instance of 'functional'.

poeschlr replied:
I am not sure your description makes more sense to be honest. Maybe i misunderstood your partial inserts.

evanshultz replied:
Yes. That could be. Let me share the entire sentence I suggest to replace the final one above: `If these non-functional parts of the MPN are at the end of the symbol name then the wildcard should be omitted.`.

Either way, there is an 'a' missing in 'functional'.

poeschlr replied:
done

evanshultz commented on content/klc/S2.3.adoc:
```diff
@@ -1,21 +1,18 @@
 +++
 
-Many electronic components are provided in multiple footprint (package) options. These _may or may not_ be pin compatible.
+Many electronic components are provided in multiple packaging variants. These _may or may not_ be pin compatible.
```
I might change 'packaging variants' to 'packages' for brevity and clarity. At least it's more clear to me.

poeschlr replied:
done

evanshultz replied:
'packaging variant' is also used on line 7 and could go to 'package'.

poeschlr replied:
done

evanshultz commented on content/libraries/klc_history.adoc:
```diff
@@ -5,6 +5,12 @@ url = "/libraries/klc/history/"
 ---
+== 3.0.21 - 2019-08-08
+* Clarified KiCad footprint assignment work-flows as generic vs fully specified symbols (G2.1)
+  * Avoid using the word atomic as it does not really specify what is really used.
```
Perhaps change the final word, 'used' to 'meant'? Also, this is the only bullet to end with a period.

poeschlr replied:
completely reworded it instead of changing only one word

evanshultz commented on content/klc/S2.3.adoc:
```diff
@@ -1,21 +1,18 @@
-Many electronic components are provided in multiple footprint (package) options. These _may or may not_ be pin compatible.
+Many electronic components are provided in multiple packaging variants. These _may or may not_ be pin compatible.
+
+KiCad is not able to assign different footprints for aliases. Fully specified symbols therefore require a separate symbol for every package variant. Similarly kicad is not able to define different 3d models other than having a separate footprint. This means a part needs a separate footprint even if the land pattern is the same if the 3d model has functional differences (example different height).
```
Same comment as above. Add a comma after 'Similarly'. There are missing capital letters in 'KiCad' and '3D'. I might even re-write the last two sentences to 'Similarly, KiCad a footprint has a 1:1 relationship with its 3D model. This means a multiple footprints are needed where the 3D model has functional differences even if the footprints are identical (for example when 3D model height differs).'.

poeschlr replied:
Reworded inspired by your suggestion but removed some things that looked wrong to me.

poeschlr commented:
I think i have again addressed everything that was pointed out.

evanshultz commented:
I left a few more comments. And the rev still needs to bump to 3.0.22.

poeschlr commented:
I will make the version bump during the merge in whichever of the two open pull requests is merged later.

evanshultz commented:
FYI, I found the original post talking about symbol naming with the package type in the suffix: https://github.com/KiCad/kicad-library/issues/1398#issuecomment-316121470.

poeschlr commented:
I think i have addressed everything. Including the version number (as part of the merge as already suggested above)

nickoe commented:
@poeschlr OK, it looks like github says there are conflicts so I can't mere it right now

poeschlr commented:
@nickoe i see no merge conflict displayed. I only have "1 change requested" which is your request regarding the version number.

nickoe commented:
> This branch cannot be rebased due to conflicts

But I do. Could you please rebase it against master?

poeschlr commented:
This is what i see:
![Screenshot from 2019-09-20 10-57-41](https://user-images.githubusercontent.com/18327350/65313915-8507c600-db95-11e9-8b34-236d59cfbf0d.png)

nickoe commented:
Well, it can be squashed, not rebased it seems.

poeschlr commented:
This is most likely because i merged master into this branch (which i needed to do because the history page was changed in quite a few other pull requests.)

evanshultz commented on content/klc/S2.1.adoc:
```diff
@@ -2,15 +2,15 @@
 +++
 
-. Library name should not be duplicated in symbol name
+. Library naming should not be duplicated in footprint name
```
This section (S2.1) is about symbols. Was it intentional to change `symbol` to `footprint`? Also, I like changing the original `Library name` to `The library name` rather than the current change.

poeschlr replied:
definitely a mistake.

evanshultz commented on content/libraries/klc_history.adoc:
```diff
@@ -5,6 +5,12 @@ url = "/libraries/klc/history/"
 
 ---
+== 3.0.22 - 2019-09-20
+* Clarified KiCad footprint assignment work-flows as generic vs fully specified symbols (G2.1)
```
Is this hyphen required? Do you want to hyphenate `fully-specified` or leave it as two separate words?

poeschlr replied:
no

evanshultz commented on content/libraries/klc_history.adoc:
```diff
@@ -5,6 +5,12 @@ url = "/libraries/klc/history/"
 ---
+== 3.0.22 - 2019-09-20
+* Clarified KiCad footprint assignment work-flows as generic vs fully specified symbols (G2.1)
+  * Replace the word atomic with fully specified as this term better fits the used methodology.
```
This bullet is the only one ending in a period. Also, can you capture the KLC clauses that were changed for each bullet?

poeschlr replied:
done

evanshultz commented:
Added onto an existing discussion with one more comment where I suggested a further edit.

I then added one comment for S2.1 and a couple more in the KLC change log.

evanshultz commented:
This PR missed updating the KLC version at https://github.com/KiCad/kicad-website/blob/master/content/libraries/klc.adoc, so the KLC version on the main KLC page doesn't reflect the content updates here.

Go to http://kicad-pcb.org/libraries/klc/ and you will still see 3.0.21 even though the changelog and content reflects the changes implemented here for 3.0.22.

Does anyone know if there's a good way to keep the KLC version on multiple pages in sync? I assume there must be but I'm not knowledgeable about this topic.

poeschlr commented:
I noticed that when i made #439
